### ISSUE TITLE
[D3D8]Top Spin 2004: career mode graphics corruptions

### ISSUE THREAD

**Issue by tomnic79** [doitsujin/dxvk#4214]
Top Spin 2004, default configuration, 1024x768, 1920x1080, 1680x1050 all @ 60 Hz. XP SP2 compatibility mode on, run as admin. Both original and with official game patch.

- GPU: nVidia GeForce RTX 3070 8 Gb (but also with a GeForce GTX 750Ti and a GeForce RTX 2080Ti)
- Driver: 560.70
- DXVK version: 2.4
- Windows 10 and 11 latest versions

Apitrace file: https://mega.nz/file/1jZHnb6Z#fUkahYqxAwjGRH74I2sVoBtM2nOCvREDe54kcMan_1s

![1](https://github.com/user-attachments/assets/3ded696d-34fc-403d-b1b0-1798a6dd1e9b)
![2](https://github.com/user-attachments/assets/999bd8f9-1f45-40b5-a9e4-d20ea2613cc3)

As you can see there is an issue rendering some poligons: totally missing or with crazy vertex positions. 

I've tried the old D8VK 1.0 with the conf option **d3d8.managedBufferPlacement = True** and it solves the problem, BUT the game exits after 2-3 minutes of gameplay without a specific error: Windows Event Manager says generically that either Topspin.exe or msvcirt.dll crashed.

Is this a possible regression from D8VK? Where can I find an equivalent conf option for DXVK 2.4 for d3d8.managedBufferPlacement = True?

Thanks in advance for your help,
Nicola

**Comment by Blisto91** [comment]
@tomnic79 Hi there. Can you check what happens if you try with `d3d9.allowDirectBufferMapping = False`

**Comment by tomnic79** [comment]
Hi @Blisto91, just tried now: no corruptions during player personalization, but hangs and exits to desktop as soon as I try to save the player, d3d8.dll causing the issue according to windows event manager.

**Comment by dungeon007** [comment]
Nah, this app case consistently leak memory right from the start or better to say leak comes from a index buffer, from a second CPU thread...
 You probably dont need even to play a game. 

![leak](https://github.com/user-attachments/assets/7744b750-e7af-455e-9d54-e3e568a2028f)

 Just launch it to this and practically do nothing there - just wait and instead watch how sys mem consistently grows up step by step to about 2GiB limit, then OOM happens of course. It goes out in about 1 minute for me. 
  Basically it leaks buffers - that is a main bug. You can ignore possible corruptions/glitches that happen afterwards.

**Comment by WinterSnowfall** [comment]
>  as of course there will be weird things since it is leaking.

Not releasing objects doesn't generally cause graphical artifacting, so I'm not sure what you mean here. The two issues are most likely unrelated.

> There were many buffer leaks with other apps on a earlier d8vk, but this case seems survived.

Previous leaks were caused by incorrect refcounting, but that doesn't seem to be the case here - or at least not the obvious ones, since refcounts for all d3d8 objects are identical between a dxvk trace and a WineD3D trace. I'll look closer into what's causing the leaks, but since the problem is rather obscure it will probably take a while.

P.S.: As a fun aside, the game outright crashes when attempting to use d3d8to9, with either dxvk or WineD3D.

**Comment by dungeon007** [comment]
For me it works with WINED3D, works with d3d8to9 to WINED3D, works with d3d8to9 DXVK's d3d9 on top of that, works with NINE too, etc... no leak nor such corruptions with any of these combinations.
![NINE](https://github.com/user-attachments/assets/410f5ab1-a369-4aa6-a9a7-04e55564f1d4)
![DXVK](https://github.com/user-attachments/assets/9ffe767e-8dd6-4ae0-a2ad-d56514418e1e)
 Seems works with whatever, sans not for long on DXVK's d3d8... who knows why, maybe because - it is not nVidia :dancers:

**Comment by tomnic79** [comment]
Yes with AMD gpus everything's fine, the problem is with NVIDIA GPUS even in native Windows without any sort of "workaround", that's why I'm trying to use some wrapping to try to solve this issue.

**Comment by WinterSnowfall** [comment]
> that's why I'm trying to use some wrapping to try to solve this issue on my nVidia based PCs

We'll probably end up applying `d3d9.allowDirectBufferMapping = False` to address the first issue. As to the memory leak, that might prove a bit more challenging to pinpoint. I don't see any obvious wrongdoings with buffers or anything else.

**Comment by dungeon007** [comment]
> Yes with AMD gpus everything's fine, the graphics corruptions problem is with nVIDIA GPUS even in native Windows without any sort of "workaround", that's why I'm trying to use some dll wrapping to try to solve this issue on my nVidia based PCs without exchanging GPUs. The freezing problem, generated by the buffer leak, happens only when wrapping with D8VK and DXVK.

Maybe try this build [d3d8.zip](https://github.com/user-attachments/files/16736222/d3d8.zip) of d3d8to9 1.9.2 (particular version where game AA is actually working) and also it is helding up index buffer. Since index is leaking, that is my quickest guess.

**Comment by tomnic79** [comment]
> > Yes with AMD gpus everything's fine, the graphics corruptions problem is with nVIDIA GPUS even in native Windows without any sort of "workaround", that's why I'm trying to use some dll wrapping to try to solve this issue on my nVidia based PCs without exchanging GPUs. The freezing problem, generated by the buffer leak, happens only when wrapping with D8VK and DXVK.
> 
> Maybe try this build [d3d8.zip](https://github.com/user-attachments/files/16736222/d3d8.zip) of d3d8to9 1.9.2 (particular version where game AA is actually working) and also it is helding up index buffer. Since index is leaking, that is my quickest guess. Probably wont work, as you said up there this.
> 
> > 've tried the old D8VK 1.0 with the conf option d3d8.managedBufferPlacement = True and it solves the problem, BUT the game exits after 2-3 minutes of gameplay without a specific error:
> 
> But it dont hurt to try... or maybe try to limit app to 1 CPU core. That also might not help, because VK is threaded by design... but again it dont hurt to try. Just to know what triggers it and if anything changes.

With your d3d8.dll version it crashes as soon as I try to create a new player, and if I try to set the affinity to CPU0 only it crashes :(

**Comment by dungeon007** [comment]
I dont think that build makes it crash, but probably these leftover temporary files from before crashes. 
 Think like this about this app **if it crashes ONCE it will crash AGAIN**... because if it crashes once, then there are leftovers that could make  further crashes. If game happens to crashed once, then remove entire **Media\GamePart\Players** folder, before running  a game again.

**Comment by tomnic79** [comment]
No it crashes the same, even exibition doesn't work anymore with the specific dll provided, nor folder Media\GamePart\Players gets recreated, unless deleting this d3d8.dll

**Comment by dungeon007** [comment]
Players folder get recreated, only at the moment right about moment your first screen that you posted.

**Comment by tomnic79** [comment]
Just to be clear that I've cleaned all the folders you've said with the d3d8.dll you suggested: https://mega.nz/file/Zv5zUQRY#bVIYjY4jAg-ANOWD_tjMdpYU9DS4RiNzMOwMrrHN9J8

You can also see that when I'm trying to start to play it hangs just before recreating the player.ini file you've just said.

The only way I can play this game at the moment with nVidia cards is with the latest dgVoodoo2 forcing DX12 render pathway (or the software based one, albeit a bit slow @ FHD). Just an issue with the intro videos that are not stretched to fill the screen (other issues if I select "stretch" in directx settings so I can live without intro videos ;) ), with D8VK videos were ok, so I'd have preferred that way.

Anyway, I'll use dgVoodoo2 or the other PCs with AMD cards, thanks for all the efforts till today!!!

**Comment by dungeon007** [comment]
>Just to be clear that I've cleaned all the folders you've said with the d3d8.dll you suggested: https://mega.nz/file/Zv5zUQRY#bVIYjY4jAg-ANOWD_tjMdpYU9DS4RiNzMOwMrrHN9J8
That looks like breaking on native nvidia d3d9 driver. 
And what about that same d3d8.dll, but put there also just DXVK's d3d9.dll too? That is immediately different path as it goes to VK driver

**Comment by tomnic79** [comment]
It doesn't crash but players are mostly black, no vertex corruptions nor missing polygons though!

**Comment by dungeon007** [comment]
Cool, try helding vertex too... well try all 4 combinations and see if any could make nvidia driver happy there 
[d3d9_buffers.zip](https://github.com/user-attachments/files/16756856/d3d9_buffers.zip)

Again, just do not forget that app requires clean exit, so that it could behave normal next time  So, no crashes, kills, ALT+F4, etc... otherwise consider it to be broken and that require some clean up :dancers:

**Comment by tomnic79** [comment]
Only with the d3d8.dll file found in default subfolder in your zip alone the game doesn't crash but players remain mostly black. Otherwise it crashes before creating a new player, just like with the first d3d8.dll you sent me.

**Comment by mrdeathjr28** [comment]
In my case work with all dxvk 2.4 files and i dont see any chacrecter rendering issue (amd renoir igpu)

![tpspa](https://github.com/user-attachments/assets/df12f077-f0e3-4f38-a003-4cff7a5aa4e4)

![tpsnb](https://github.com/user-attachments/assets/c106bc75-8af6-4702-ad80-935d7de2db3d)

![tpspc](https://github.com/user-attachments/assets/2cae5873-f583-4e2d-ad94-2fc7761d39b1)

**Comment by tomnic79** [comment]
> In my case work with all dxvk 2.4 files and i dont see any character rendering issue (amd renoir igpu)
> 
> ![tpspa](https://private-user-images.githubusercontent.com/6601480/361940566-df12f077-f0e3-4f38-a003-4cff7a5aa4e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3ODU4OTksIm5iZiI6MTcyNDc4NTU5OSwicGF0aCI6Ii82NjAxNDgwLzM2MTk0MDU2Ni1kZjEyZjA3Ny1mMGUzLTRmMzgtYTAwMy00Y2ZmN2E1YWE0ZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMTkwNjM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTJkNjA2ZDU4YTU5ODliYWZlNDYxMmVmMTE4YTU0NjI0MzkzMzVjYzI1OWRhM2I4YmNjNjUxNzBhY2Q4ZGY0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LMh5S7UEQOeSwSFOMbGW8ouhHEUbnBX1dAu1lT_6j1M)
> 
> ![tpsnb](https://private-user-images.githubusercontent.com/6601480/361940611-c106bc75-8af6-4702-ad80-935d7de2db3d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3ODU4OTksIm5iZiI6MTcyNDc4NTU5OSwicGF0aCI6Ii82NjAxNDgwLzM2MTk0MDYxMS1jMTA2YmM3NS04YWY2LTQ3MDItYWQ4MC05MzVkN2RlMmRiM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMTkwNjM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTEyMjNjOGIyMzI3NDBiYmJjNDVjOGViM2JjNTk0M2ExODk3YTUwZTlhNDE0NmQwYjI5ZDc2MzUxMjEwNWM2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.hVz9bhy4Yr7LmN9p1sJFcra6d2Nm26hLdFhZ46qn4A0)
> 
> ![tpspc](https://private-user-images.githubusercontent.com/6601480/361940646-2cae5873-f583-4e2d-ad94-2fc7761d39b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3ODU4OTksIm5iZiI6MTcyNDc4NTU5OSwicGF0aCI6Ii82NjAxNDgwLzM2MTk0MDY0Ni0yY2FlNTg3My1mNTgzLTRlMmQtYWQ5NC0yZmM3NzYxZDM5YjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMTkwNjM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmU3MTBhNzE1NmU2ODk3NTYwN2Y1ZGNkZjhjNmQ0MThmODZhOGI0ZGUwMmJlOWIyYzBhMmZlMmRmYWExZjdiYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ygJrOSyPhnEJtlsFybl-UXqW3RnpoEy8ibone20j5-U)

Yes with AMD GPUs no issues at all, we need to solve this problem for nVidia owners

**Comment by mrdeathjr28** [comment]
> > In my case work with all dxvk 2.4 files and i dont see any character rendering issue (amd renoir igpu)
> > ![tpspa](https://private-user-images.githubusercontent.com/6601480/361940566-df12f077-f0e3-4f38-a003-4cff7a5aa4e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3ODU4OTksIm5iZiI6MTcyNDc4NTU5OSwicGF0aCI6Ii82NjAxNDgwLzM2MTk0MDU2Ni1kZjEyZjA3Ny1mMGUzLTRmMzgtYTAwMy00Y2ZmN2E1YWE0ZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMTkwNjM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTJkNjA2ZDU4YTU5ODliYWZlNDYxMmVmMTE4YTU0NjI0MzkzMzVjYzI1OWRhM2I4YmNjNjUxNzBhY2Q4ZGY0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LMh5S7UEQOeSwSFOMbGW8ouhHEUbnBX1dAu1lT_6j1M)
> > ![tpsnb](https://private-user-images.githubusercontent.com/6601480/361940611-c106bc75-8af6-4702-ad80-935d7de2db3d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3ODU4OTksIm5iZiI6MTcyNDc4NTU5OSwicGF0aCI6Ii82NjAxNDgwLzM2MTk0MDYxMS1jMTA2YmM3NS04YWY2LTQ3MDItYWQ4MC05MzVkN2RlMmRiM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMTkwNjM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTEyMjNjOGIyMzI3NDBiYmJjNDVjOGViM2JjNTk0M2ExODk3YTUwZTlhNDE0NmQwYjI5ZDc2MzUxMjEwNWM2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.hVz9bhy4Yr7LmN9p1sJFcra6d2Nm26hLdFhZ46qn4A0)
> > ![tpspc](https://private-user-images.githubusercontent.com/6601480/361940646-2cae5873-f583-4e2d-ad94-2fc7761d39b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3ODU4OTksIm5iZiI6MTcyNDc4NTU5OSwicGF0aCI6Ii82NjAxNDgwLzM2MTk0MDY0Ni0yY2FlNTg3My1mNTgzLTRlMmQtYWQ5NC0yZmM3NzYxZDM5YjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjdUMTkwNjM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmU3MTBhNzE1NmU2ODk3NTYwN2Y1ZGNkZjhjNmQ0MThmODZhOGI0ZGUwMmJlOWIyYzBhMmZlMmRmYWExZjdiYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ygJrOSyPhnEJtlsFybl-UXqW3RnpoEy8ibone20j5-U)
> 
> Yes with AMD GPUs no issues at all, we need to solve this problem for nVidia owners

only for give more feedback (amd renoir in this case) and i use new wow64 mode 
(./configure --enable-archs=i386,x86_64)

**Comment by dungeon007** [comment]
>Only with the d3d8.dll file found in the default subfolder of your zip alone the game doesn't crash but **players remain mostly black**. Otherwise it crashes before creating a new player, just like with the first d3d8.dll you sent me.

That alone could be just some formats issue too. Try running app in a window too, driver might pick some else formats there, depending if app is window or if fulscreen is normal or borderless or exclusive or whatever.
 There is no option where user could specify 16 or 32bit, so that could go somewhat random of what driver decide.

**Comment by WinterSnowfall** [comment]
I've captured [a trace of the demo with WineD3D](https://drive.proton.me/urls/2YMXY5HMMC#BoXLSrY242rJ), however it crashes my Nvidia GPU (Xid 13) on playback with dxvk. 

```
NVRM: Xid (PCI:0000:01:00): 13, pid='<unknown>', name=<unknown>, Graphics Exception on GPC 0: 3D HEIGHT CT Violation. Coordinates: (0x2, 0x208)
NVRM: Xid (PCI:0000:01:00): 13, pid='<unknown>', name=<unknown>, Graphics Exception on GPC 3: WIDTH CT Violation. Coordinates: (0x280, 0xd)
```

Playing it back with WineD3D shows some warning messages, but works fine.

Not entirely sure if this is a problem with apitrace or the game, but I've asked @Blisto91 to try replaying it with Vulkan validation layer enabled, see if that yields anything interesting.

**Comment by dungeon007** [comment]
Noticed that dgVoodoo2 (ver 81_3) in d3d12_fl11_0 mode crashes in msvcr71 by running on newest vkd3d-proton, but runs on some earler versions of vkd3d-proton from last year.
 Thats cool, so lets bisect this, might yield what is this about here too...

**Comment by WinterSnowfall** [comment]
@tomnic79 Sorry that it took so long to fix, but the game proved quite difficult to procure. It should work fine now with this build, please try it out: https://github.com/doitsujin/dxvk/actions/runs/17040289378/artifacts/3788253188

**Comment by GOOGLEQuyloon** [comment]
Hello. I copied the .dlls into the root directory of the game's folder. The files did nothing to help fix Top Spin (PC)..

**Comment by WinterSnowfall** [comment]
Note that you need both the 32-bit d3d8.dll and d3d9.dll files. Otherwise, the game is known to work fine on latest dxvk. If you are still having issues with it for some reason, please open a new issue and fill out the template to give us more details, so that we can look into it.

**Comment by GOOGLEQuyloon** [comment]
Hello. Thank you. From what we read, we are simply taking the said two .dll files and putting them in the root directory of the main folder of the application of interest; correct? There is also no need to run another application to make the .dlls work? We already know the answer to those questions but just want confirmation. Thank you very much!

We tried all combinations of copying and pasting the .dll files into the root directory when nothing work. When we say nothing, we mean nothing! In fact, the 32-bit .dlls were even more tragic for us. Why? The application would not even open, it would crash in career mode, it would crash in exhibition mode!

Guess what? We tried the 32-bit .dll files first like you said before putting our comment here. We exhausted all means and combinations of 32-bit and 64-bit dll files (just to try at this point). 

Again, the application crashed with the 32-bit dlls. At the mercy of just trying, we took the 64-bit .dll files, and the application worked!! Yet, please do not get too excited. It worked as in it allowed us to at least open and run the game and even not crash once at all during the career mode process. and all in-app user's interfaces.

Yet, the character bugs remained. We can see the Z formations in the imagery and can also see with the 64-bit dlls that something is trying to form?! (faring on the positive side of things).

Insight: We are doing a seven continental statistical study on sports behavioral outcomes, among others, that Tennis Nation's and ESPN's commentator, and professional coach Brad Gilbert even liked and commented on years ago. You may Google Quyloon for more information as to these claims.

However, it is kind of hard to do that if all we see are shadow players; which is weird because we can see the top sixteen standards well. However, it is not the case for the others. We are thinking that their data files have been written differently. Especially, we see this as the case with more updated computer systems and devices. 

Well, at least, it adds to the level of difficulty for us playing with a no skill, no star created player where we cannot see them but only their shadow(s). Yes. We created a tour of our own! That makes it very difficult to play and win, especially on the Vienna court (our only loss since 2025). Oh, we also are not allowed to use risk shots and try to also avoid using drop shots. In an exhibition match, we FINALLY beat the other worldly Martina Hingis who has 15 stars and 4 skills to our 0 stars, 0 skills, no risk shots, no spoon serves, and minimum to no drop shots allowed rules.

We won 20 titles by now (one Australian Open 2025 and 15 other titles of varying tiers; 1000 masters, 500 major pro, 250 Pro-Am, 250 minor, and 100 future). We had to shorten the 2025 season of some 45 different events of 16 participants of various skill levels due to other volunteer projects we are also working on in upstate New York, volunteering for other branch locations while also assisting those doing undisclosed, unpaid work.

This 2026 season, so far, we won four titles (1 at the 250 level and three at the 100 level). We are now in the 500 event.

Thank you for your quick response. We also hope that the added information provides insight as to why we are hoping that the .dlls that are marketed as working actually work! We would so much appreciate that.

**Comment by WinterSnowfall** [comment]
> Hello. Thank you. From what we read, we are simply taking the said two .dll files and putting them in the root directory of the main folder of the application of interest; correct?

Copying them next to the game executable usually works well enough. That may or may not be the "root" directory of the application.

> Again, the application crashed with the 32-bit dlls. At the mercy of just trying, we took the 64-bit .dll files, and the application worked!! Yet, please do not get too excited. It worked as in it allowed us to at least open and run the game and even not crash once at all during the career mode process. and all in-app user's interfaces.

The 64-bit dlls won't do anything (won't even load), as the application is 32-bit in this particular case.

> Thank you for your quick response. We also hope that the added information provides insight as to why we are hoping that the .dlls that are marketed as working actually work! We would so much appreciate that.

As I've previously stated, please open a new issue and fill in the issue template, attaching generated logs and giving us information about your OS, hardware, drivers etc. It's highly likely the issue here is something specific to your setup though.

**Comment by GOOGLEQuyloon** [comment]
Yes. Okay. We are telling you that the 32-bit dll were problematic for us, whereas the 64-bit dll files were not. That is our findings.

We would like to recreate a case. However, the "Reopen issue" button does not work for us. In fact, when we hover over the said button, it states, "You do not have permissions to reopen this issue."

Thank you so much for your time.
